src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final String DEFAULT_COLOR = Color.WHITE.name();
+    private static final double RADIUS = 10;
+    private ColorSupplier color = new ColorSupplier();
+    private RandomDoubleGenerator randomDouble = new RandomDoubleGenerator();
+    private Random random = new Random();
+
+    private Circle getRandomCircle() {
+        return new Circle(color.getRandomColor(), randomDouble.getRandomDouble());
+    }
+
+    private Square getRandomSquare() {
+        return new Square(color.getRandomColor(), randomDouble.getRandomDouble());
+    }
+
+    private Rectangle getRandomRectangle() {
+        return new Rectangle(color.getRandomColor(), randomDouble.getRandomDouble(),
+                randomDouble.getRandomDouble());
+    }
+
+    private RightTriangle getRandomRightTriangle() {
+        return new RightTriangle(color.getRandomColor(), randomDouble.getRandomDouble(),
+                randomDouble.getRandomDouble());
+    }
+
+    private IsoscelesTrapezoid getRandomIsoscelesTrapezoid() {
+        return new IsoscelesTrapezoid(color.getRandomColor(), randomDouble.getRandomDouble(),
+                randomDouble.getRandomDouble(), randomDouble.getRandomDouble());
+    }
+
+    public Circle getDefaultFigure() {
+        return new Circle(DEFAULT_COLOR, RADIUS);
+    }
+
+    public Figure getRandomFigure() {
+        FigureName[] figureNames = FigureName.values();
+        for (FigureName figureName : figureNames) {
```

Comment:
Is this random? You will be getting the same figure each time, and won't ever actually iterate over this array. 
Use random number and FigureName.values()[number] instead